### DISCUSSION TITLE
tar_watch times out after a while

### DISCUSSION THREAD

**Discussion by mem48** [ropensci/targets#1197] [Help]
## Prework

* [x ] I understand and agree to [help guide](https://books.ropensci.org/targets/help.html).
* [x ] I understand and agree to [contributing guide](https://github.com/ropensci/targets/blob/main/CONTRIBUTING.md).
* [x ] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.

## Bug

I have a long-running build (about 24 hours) that I use `tar_watch(seconds = 60, targets_only = TRUE)` to monitor. For the first few hours everything works as expected but then the web browser times out. It stays permanently greyed out with the pulsing circle in the top left. The same thing happens when it updates every 60 seconds, except in this case it never updates.

![image](https://github.com/ropensci/targets/assets/15819577/05a11fd0-75e1-4979-8564-df9582f1f6e6)

I get a warning in the browser console

```
downloadable font: Glyph bbox was incorrect (glyph ids 1 2 3 4 5 8 9 10 11 12 13 14 16 17 19 22 24 28 32 34 35 38 39 40 43 44 45 46 47 50 51 52 53 55 56 58 60 61 62 64 67 68 70 71 72 73 74 78 79 80 81 83 90 96 101 103 104 105 108 109 115 116 117 118 120 123 125 135 138 139 140 142 143 144 145 146 148 149 154 155 157 161 162 163 164 165 169 170 171 173 179 181 193 195 203 207 208 210 211 214 218 219 223 225 227 228 229 230 235 236 237 238 239 240 245 246 247 248 249 250 251 252 253 254 255 256 257 263 264 266 268 271 275 278 279 280 281 282 283 284 285 286 287 288 291 292 293 294 295 296 297 298 299 300 301 302 303 304 305 306 307 308 309 310 311 312 313 314 315 316 321 335 338 339 340 342 344 345 346 353 354 356 357 358 359 362 363 365 366 371 373 374 379 381 382 383 386 389 390 391 393 394 406 407 412 413 418 419 420 424 432 433 439 448 449 450 451 454 455 456 457 472 479 480 481 482 485 486 490 491 493 499 500 501 503 508 509 513 515 516 525 527 528 532 535 541 542 543 549 550 551 552 554 555 556 558 560 569 571 593 602 603 604 607 608 609 614 615 617 618 623 626 627 643 644 645 647 650 651 654 655 656 657 662 663 664 665 670 671 672 674 675 679 680 681 682 683 698 699 708 712 714 717 718 729 730 732 735 736 739 746 747 752 761 762 767 774 776 777 778 779 788 789 790 794 796 798 799 800 801 803 804 806 826 828 829 831 835 836 838 839 840 841 842 843 844 845 848 849 856 857 861 862 863 871 873 874 880 882 892 895 900 908 911 913 925 928 929 930 933 936 937 938 941 942 943 944 945 948 949 950 952 958 960 961 962 964 966 967 969 973 974 978 979 980 981 982 989 998 1000 1001 1005 1006 1008 1009 1011 1012 1013 1016 1020 1026 1027 1031 1036 1037 1042 1045 1048 1050 1052 1053 1057 1058 1060 1063 1072 1073 1076 1084 1087 1099 1104 1110 1111 1112 1116 1117 1121 1122 1124 1131 1136 1140 1141 1142 1147 1148 1151 1157 1163 1167 1168 1170 1177 1186 1187 1193 1196 1199 1200 1201 1204 1205 1208 1211 1212 1217 1218 1220 1224 1226 1228 1230 1231 1232 1233 1235 1236 1237 1238 1243 1246 1247 1249 1251 1256 1258 1259 1260 1261 1265 1268 1269 1271 1272 1273 1275 1276 1279 1285 1289 1290 1291 1292 1296 1297 1303 1304 1305 1309 1310 1311 1312 1317 1319 1320 1324 1325 1328 1329 1330 1331 1334 1335 1337 1339 1341 1343 1356 1357 1363 1369 1371 1372 1375 1376 1377 1382 1384 1387) (font-family: "Font Awesome 6 Free" style:normal weight:900 stretch:100 src index:0) source: http://127.0.0.1:61109/font-awesome-6.4.2/webfonts/fa-solid-900.woff2 

```

Fortunately, the actual build continues as normal. But it would be nice to have the web UI for the whole build.

My build process is

```
library(targets)
tar_watch(seconds = 60, targets_only = TRUE)
tar_make_future(workers = 4)
```

R.Version 

```
> R.Version()
$platform
[1] "x86_64-w64-mingw32"

$arch
[1] "x86_64"

$os
[1] "mingw32"

$crt
[1] "ucrt"

$system
[1] "x86_64, mingw32"

$status
[1] ""

$major
[1] "4"

$minor
[1] "2.2"

$year
[1] "2022"

$month
[1] "10"

$day
[1] "31"

$`svn rev`
[1] "83211"

$language
[1] "R"

$version.string
[1] "R version 4.2.2 (2022-10-31 ucrt)"

$nickname
[1] "Innocent and Trusting"

```

Browser
Firefox 120.0.1 (64 Bit)

**Comment by mem48** [comment]
The full repo is here https://github.com/nptscot/npt

**Comment by wlandau** [comment]
@mem48, several hours is a long time to run a session of a Shiny app. Unless you are actively looking at the screen, you can safely close `tar_watch()` and reopen it when you want to check again. The app starts quickly enough. If you still want to run it for a long time, you could try setting `supervise = FALSE` and `poll_connection = FALSE` to make the background process more independent of its parent process and less likely to close if anything happens to the parent.

In the future, before assuming there is a bug in `targets`, please post to the discussion forum. Most reported "bugs" are user errors and platform-specific edge cases, and we cannot attribute the cause without a proper look.
